build-conflicts.php

```diff
@@ -131,6 +135,62 @@ static function ($errorCode, $message = '', $file = '', $line = 0) : void {
         ));
     };
 
+    /** @return Advisory[] */
+    $getGitHubAdvisories = static function () use ($token) : array {
```

Comment:
I'd design an interface:

```php
interface GetAdvisories
{
    /** @psalm-return \Generator<Advisory> - note: not sure what template parameters are required by Generator */
    public function __invoke(): \Generator;
}
```

Then there would be 3 implementations:

 1. `GetAdvisoriesFromFriendsOfPhpAdvisories`
 2. `GetAdvisoriesFromGithubApi`
 3. `GetAdvisoriesFromMultipleSources`

Specifically, the last one is relatively easy to build then:

```php
final class GetAdvisoriesFromMultipleSources implements GetAdvisories
{
    private $concrete;
    public function __construct(GetAdvisories ...$concrete) { $this->concrete = $concrete; }
    public function __invoke() {
        foreach ($this->concrete as $concrete) {
            yield from $concrete();
        }
    }
}
```

Reply:
will implement this.   
I was wondering If I should implement client myself and have no dependencies or install a package, dunno which one tbh

Reply:
I endorse the `php-http` packages (https://packagist.org/packages/php-http/httplug)

Reply:
thanks, will use it :+1: